earth_enterprise/src/fusion/autoingest/sysman/AssetVersionD.cpp

```diff
@@ -215,52 +227,156 @@ AssetVersionImplD::SyncState(void) const
     AssetDefs::State newstate = ComputeState();
     if (newstate != state) {
       MutableAssetVersionD self(GetRef());
-      self->SetState(newstate);
+      self->SetState(newstate, true, notifySet);
     }
   } else {
     AssetDefs::State newstate = ComputeState();
     if (newstate != state) {
       MutableAssetVersionD self(GetRef());
-      self->SetState(newstate);
+      self->SetState(newstate, true, notifySet);
     }
   }
 }
 
+AssetVersionImplD::NotifySet * AssetVersionImplD::getNotifySet(NotifySet * const notifySet) {
+  if (!MiscConfig::Instance().ConsolidateListenerNotifications) {
+    return nullptr;
+  }
+  else if (notifySet) {
+    return notifySet;
+  }
+  else {
+    return new NotifySet();
+  }
+}
+
+void AssetVersionImplD::HandleNotifySet(NotifySet * const notifySet, NotifySet * const myNotifySet) {
+  if (!notifySet && myNotifySet) {
+    PropagateStateChangeToParentsAndListeners(myNotifySet);
+    delete myNotifySet;
+  }
+}
+
 void
-AssetVersionImplD::PropagateStateChange(void)
-{
-  notify(NFY_VERBOSE, "PropagateStateChange(%s): %s",
-         ToString(state).c_str(), GetRef().c_str());
-  for (std::vector<std::string>::const_iterator p = parents.begin();
-       p != parents.end(); ++p) {
-    AssetVersionD parent(*p);
+AssetVersionImplD::PropagateStateChangeToParents(
+    const std::set<std::string> & toNotify,
```

Comment:
Similar comment as above.  Variable names `noNotify` and `notifySet` lack descriptiveness, and it's not clear why someone trying to propagate a state change to parents needs to know about notify sets and their details.